hostplumber/pkg/hoststate/hoststate.go

```diff
@@ -134,10 +135,25 @@ func (hni *HostNetworkInfo) addNetPciDevice(devicePath, ifName string) error {
 	}
 	ifStatus.MTU = mtu
 
+	ipv4Addrs, err := iputils.GetIpv4Cidr(ifName)
+	if err != nil || len(*ipv4Addrs) == 0 {
```

Comment:
You are only using ipv4Addrs by converting to the value. Maybe just return a (non-pointed to) slice instead. (see comment on the GetIpv4Cidr)